### PR TITLE
Extract transit tech stacks airtable

### PR DESCRIPTION
**Pull request by lauriemerrell**
_🚨 Note that this is merging into `airtable-updates` and not `main`_

# Description

This is another intermediate PR that will be merged into main via #1509. 

This PR:
* Extracts data from the Transit Technology Stacks Airtable base via new `airtable_loader.*` and `create_external_tables.*` Airflow tasks
* Creates staging and mart tables for the new TTS data
* Creates a macro & procedure for mapping between copies of a synced table in two different bases & implements it for `organizations` and `services` 

This PR also changes the Airtable operator naming process to guarantee that column names will be BQ-safe 👍 

After merging I am going to make a subsequent update to stop renaming the internal Airtable IDs to `<entity>_id` which will require a bit of subsequent refactoring but I decided this PR is already big enough. 

Once merged into `main` via #1509 it will address #1159

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation
- [ ] agencies.yml

## How has this been tested?

Confirmed that all of the following run without errors.

* `dbt run -s +models/mart/transit_database`
* `dbt test -s +models/mart/transit_database`
* `dbt docs generate`
* `dbt docs serve`

Also confirmed that the new DAG tasks run successfully in local Airflow.

### CODE REVIEW

**Review by atvaccaro — CHANGES_REQUESTED** [review]

File: warehouse/models/mart/transit_database/dim_contract_attachments.sql [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+
+dim_contract_attachments AS (
+    SELECT
+        unnested_attachments.id AS key,
+        stg_transit_database__contracts.key AS contract_key,
+        stg_transit_database__contracts.name AS contract_name,
+        unnested_attachments.url AS attachment_url,
+        stg_transit_database__contracts.calitp_extracted_at
+    FROM stg_transit_database__contracts,
+        stg_transit_database__contracts.attachments AS unnested_attachments
```
[the remark this answers is not in the document]
Comment (lauriemerrell) [reply warehouse/models/mart/transit_database/dim_contract_attachments.sql:15]: Implicit cross join: https://cloud.google.com/bigquery/docs/reference/standard-sql/arrays#querying_array-type_fields_in_a_struct. If you think it needs to be explicit I can make it so

File: warehouse/models/staging/transit_database/int/int_tts_services_ct_services_map.sql [hunk 2]
```diff
@@ -0,0 +1,32 @@
+WITH
+
+ct_services AS (
+    SELECT *
+    FROM {{ source('airtable', 'california_transit__services') }}
+    WHERE TRIM(name) IS NOT NULL
```
[the remark this answers is not in the document]
Comment (lauriemerrell) [reply warehouse/models/staging/transit_database/int/int_tts_services_ct_services_map.sql:6]: ope good call, will fix

File: warehouse/macros/get_latest_external_data.sql [hunk 3]
```diff
@@ -0,0 +1,10 @@
+{% macro get_latest_external_data(external_table_name, columns) %}
+
+WITH ranked AS (
+    SELECT *,
+        DENSE_RANK() OVER (ORDER BY {{ columns }}) as rank
```
[the remark this answers is not in the document]
Comment (lauriemerrell) [reply warehouse/macros/get_latest_external_data.sql:5]: There is no partition (we want to rank the whole table) and we need to allow ties -- we want to grab all the rows that were in the most recent extract. So `DENSE_RANK` actually matters. I can update the columns parameter name though to be a bit more explicit

File: warehouse/models/staging/transit_database/stg_transit_database__services.sql [hunk 4]
```diff
@@ -0,0 +1,25 @@
+{{ config(materialized='table') }}
+
+WITH
+latest AS (
+    {{ get_latest_external_data(
+        external_table_name = source('airtable', 'california_transit__services'),
+        columns = 'dt DESC, time DESC'
```
[the remark this answers is not in the document]
Comment (lauriemerrell) [reply warehouse/models/staging/transit_database/stg_transit_database__services.sql:7]: updated to be called `order_by`, see above

File: warehouse/macros/transit_database_synced_table_id_mapping.sql [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
+    table_b,
+    base_b,
+    table_b_id_col,
+    table_b_name_col,
+    table_b_date_col,
+    shared_id_name,
+    shared_name_name,
+    shared_date_name) %}
+
+   SELECT DISTINCT
```
[the remark this answers is not in the document]
Comment (lauriemerrell) [reply warehouse/macros/transit_database_synced_table_id_mapping.sql:16]: added as SQL comment, not sure of best way to format? LMK

**Review by atvaccaro — APPROVED** [review]

**Pull request merged**